service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -317,10 +329,47 @@ protected PipelineRun writePipelineRunToDbThrowsDuplicateException(PipelineRun p
     return pipelineRun;
   }
 
+  private boolean pipelineRunExistsWithJobId(UUID jobId) {
+    return pipelineRunsRepository.existsByJobId(jobId);
+  }
+
   public PipelineRun getPipelineRun(UUID jobId, String userId) {
     return pipelineRunsRepository.findByJobIdAndUserId(jobId, userId).orElse(null);
   }
 
+  /**
+   * Mark a pipelineRun as RUNNING in our database and store the user-provided job description and
+   * resultUrl.
+   *
+   * <p>We check that the pipelineRun already exists in our database and that the existing
+   * pipelineRun has status PREPARING.
+   *
+   * @param jobId
+   * @param userId
+   * @param description
+   * @param resultUrl
+   * @return pipelineRun
+   */
+  public PipelineRun startPipelineRunInDb(
+      UUID jobId, String userId, String description, String resultUrl) {
+    PipelineRun pipelineRun = getPipelineRun(jobId, userId);
+    if (pipelineRun == null) {
+      throw new BadRequestException(
+          "JobId %s not found. You must prepare a pipeline run before starting it."
+              .formatted(jobId));
+    }
+    // only allow starting a pipeline run if it is in the PREPARING state
+    if (!pipelineRun.getStatus().equals(CommonPipelineRunStatusEnum.PREPARING.toString())) {
+      throw new BadRequestException(
+          "JobId %s is not in the PREPARING state. Cannot start pipeline run.".formatted(jobId));
+    }
+    pipelineRun.setStatus(CommonPipelineRunStatusEnum.RUNNING.toString());
```

Comment:
this column should also get the `Enum` Treatment to make this less annoying. If you wnat to add it to the ticket you created for that other column we can punt on this.

Reply:
good call, updated https://broadworkbench.atlassian.net/browse/TSPS-263